File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/composite/File.java
package patterns.composite;

public class File extends Node {
    public File(String name, long size) {
        super(name);
        this.size = size;
    }
    public String toString() {
        return name + "(" + size + ")";
    }
    public long length() {
        return size;
    }
    protected void print(int depth) {
        printPrefix(depth);
        System.out.println(this);
    }
    private long size;
}

File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Composite/File.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Composite {
    internal class File : Node {
        public File(string name, long size) : base(name) {
            this.size = size;
        }

        public override String ToString() {
            return name + "(" + size + ")";
        }

        public override long Length {
            get { return size;  }
        }

        public override void Print(int depth) {
            PrintPrefix(depth);
            Console.WriteLine(this);
        }

        private readonly long size;
    }
}